trace/pool.go

```
@@ -110,22 +110,28 @@ func getPool(ctx context.Context) *poolCtx {
 	return c
 }
 
-func tryUnblock(tr *trace, pctx *poolCtx) {
+//tryUnblockWithLock will attempt to compile any LHInstructions in the trace
+func (pctx *poolCtx) tryUnblockWithLock(tr *trace) {
+	if pctx == nil {
```

hmm, we didn't have these extra checks before - in the old version, if pctx was nil, it would have just blown up. What's motivating these extra checks? (same in decrement)

---

trying to avoid blowing up ;)
thought the convention was that functions should tolerate nil receivers? (though there's lots of places were we don't follow that)

---

Unless it's explicitly required, I would avoid trying to cope with nil receivers - to me it seems dangerous to have code that you would expect to do something turn into a noop because of a mistake further up the call stack - i.e. I would have thought that most of the time, a nil receiver is a mistake, and certainly I tend to code for it rarely (though it is useful from time to time). Given how the code was before, unless we now know that we should be coping with these methods being called with a nil pool, I don't see why we would attempt to silently swallow such a situation.

---

yeah, guess the question to ask is "does nil have a meaning here?"